// formik-material-ui-pickers/dist/index.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var React = require('react');
var pickers = require('@material-ui/pickers');
var formik = require('formik');

/*! *****************************************************************************
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the Apache License, Version 2.0 (the "License"); you may not use
this file except in compliance with the License. You may obtain a copy of the
License at http://www.apache.org/licenses/LICENSE-2.0

THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
MERCHANTABLITY OR NON-INFRINGEMENT.

See the Apache Version 2.0 License for specific language governing permissions
and limitations under the License.
***************************************************************************** */

var __assign = function() {
    __assign = Object.assign || function __assign(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};

function __rest(s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) if (e.indexOf(p[i]) < 0)
            t[p[i]] = s[p[i]];
    return t;
}

function createErrorHandler(fieldError, fieldName, setFieldError) {
    return function (error) {
        if (error !== fieldError && error !== '') {
            setFieldError(fieldName, error ? String(error) : undefined);
        }
    };
}

function fieldToDatePicker(_a) {
    var _b = _a.field, _onChange = _b.onChange, fieldOnBlur = _b.onBlur, field = __rest(_b, ["onChange", "onBlur"]), _c = _a.form, isSubmitting = _c.isSubmitting, touched = _c.touched, errors = _c.errors, setFieldValue = _c.setFieldValue, setFieldError = _c.setFieldError, disabled = _a.disabled, onChange = _a.onChange, onBlur = _a.onBlur, onError = _a.onError, props = __rest(_a, ["field", "form", "disabled", "onChange", "onBlur", "onError"]);
    var fieldError = formik.getIn(errors, field.name);
    var showError = formik.getIn(touched, field.name) && !!fieldError;
    return __assign(__assign({ error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled !== null && disabled !== void 0 ? disabled : isSubmitting, onChange: onChange !== null && onChange !== void 0 ? onChange : function (date) {
            setFieldValue(field.name, date);
        }, onBlur: onBlur !== null && onBlur !== void 0 ? onBlur : function (e) {
            fieldOnBlur(e !== null && e !== void 0 ? e : field.name);
        }, onError: onError !== null && onError !== void 0 ? onError : createErrorHandler(fieldError, field.name, setFieldError) }, field), props);
}
function DatePicker(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(pickers.DatePicker, __assign({}, fieldToDatePicker(props)), children));
}
DatePicker.displayName = 'FormikMaterialUIDatePicker';

function fieldToKeyboardDatePicker(_a) {
    var disabled = _a.disabled, _b = _a.field, _onChange = _b.onChange, fieldOnBlur = _b.onBlur, field = __rest(_b, ["onChange", "onBlur"]), _c = _a.form, isSubmitting = _c.isSubmitting, touched = _c.touched, errors = _c.errors, setFieldValue = _c.setFieldValue, setFieldError = _c.setFieldError, onChange = _a.onChange, onBlur = _a.onBlur, onError = _a.onError, props = __rest(_a, ["disabled", "field", "form", "onChange", "onBlur", "onError"]);
    var fieldError = formik.getIn(errors, field.name);
    var showError = formik.getIn(touched, field.name) && !!fieldError;
    return __assign(__assign({ error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled !== null && disabled !== void 0 ? disabled : isSubmitting, onChange: onChange !== null && onChange !== void 0 ? onChange : function (date) {
            setFieldValue(field.name, date);
        }, onBlur: onBlur !== null && onBlur !== void 0 ? onBlur : function (e) {
            fieldOnBlur(e !== null && e !== void 0 ? e : field.name);
        }, onError: onError !== null && onError !== void 0 ? onError : createErrorHandler(fieldError, field.name, setFieldError) }, field), props);
}
function KeyboardDatePicker(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(pickers.KeyboardDatePicker, __assign({}, fieldToKeyboardDatePicker(props)), children));
}
KeyboardDatePicker.displayName = 'FormikMaterialUIKeyboardDatePicker';

function fieldToTimePicker(_a) {
    var disabled = _a.disabled, _b = _a.field, _onChange = _b.onChange, fieldOnBlur = _b.onBlur, field = __rest(_b, ["onChange", "onBlur"]), _c = _a.form, isSubmitting = _c.isSubmitting, touched = _c.touched, errors = _c.errors, setFieldValue = _c.setFieldValue, setFieldError = _c.setFieldError, onChange = _a.onChange, onBlur = _a.onBlur, onError = _a.onError, props = __rest(_a, ["disabled", "field", "form", "onChange", "onBlur", "onError"]);
    var fieldError = formik.getIn(errors, field.name);
    var showError = formik.getIn(touched, field.name) && !!fieldError;
    return __assign(__assign({ error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled !== null && disabled !== void 0 ? disabled : isSubmitting, onChange: onChange !== null && onChange !== void 0 ? onChange : function (date) {
            setFieldValue(field.name, date);
        }, onBlur: onBlur !== null && onBlur !== void 0 ? onBlur : function (e) {
            fieldOnBlur(e !== null && e !== void 0 ? e : field.name);
        }, onError: onError !== null && onError !== void 0 ? onError : createErrorHandler(fieldError, field.name, setFieldError) }, field), props);
}
function TimePicker(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(pickers.TimePicker, __assign({}, fieldToTimePicker(props)), children));
}
TimePicker.displayName = 'FormikMaterialUITimePicker';

function fieldToKeyboardTimePicker(_a) {
    var disabled = _a.disabled, _b = _a.field, _onChange = _b.onChange, fieldOnBlur = _b.onBlur, field = __rest(_b, ["onChange", "onBlur"]), _c = _a.form, isSubmitting = _c.isSubmitting, touched = _c.touched, errors = _c.errors, setFieldValue = _c.setFieldValue, setFieldError = _c.setFieldError, onChange = _a.onChange, onBlur = _a.onBlur, onError = _a.onError, props = __rest(_a, ["disabled", "field", "form", "onChange", "onBlur", "onError"]);
    var fieldError = formik.getIn(errors, field.name);
    var showError = formik.getIn(touched, field.name) && !!fieldError;
    return __assign(__assign({ error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled != undefined ? disabled : isSubmitting, onChange: onChange !== null && onChange !== void 0 ? onChange : function (date) {
            setFieldValue(field.name, date);
        }, onBlur: onBlur !== null && onBlur !== void 0 ? onBlur : function (e) {
            fieldOnBlur(e !== null && e !== void 0 ? e : field.name);
        }, onError: onError !== null && onError !== void 0 ? onError : createErrorHandler(fieldError, field.name, setFieldError) }, field), props);
}
function KeyboardTimePicker(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(pickers.KeyboardTimePicker, __assign({}, fieldToKeyboardTimePicker(props)), children));
}
KeyboardTimePicker.displayName = 'FormikMaterialUIKeyboardTimePicker';

function fieldToDateTimePicker(_a) {
    var disabled = _a.disabled, _b = _a.field, _onChange = _b.onChange, fieldOnBlur = _b.onBlur, field = __rest(_b, ["onChange", "onBlur"]), _c = _a.form, isSubmitting = _c.isSubmitting, touched = _c.touched, errors = _c.errors, setFieldValue = _c.setFieldValue, setFieldError = _c.setFieldError, onChange = _a.onChange, onBlur = _a.onBlur, onError = _a.onError, props = __rest(_a, ["disabled", "field", "form", "onChange", "onBlur", "onError"]);
    var fieldError = formik.getIn(errors, field.name);
    var showError = formik.getIn(touched, field.name) && !!fieldError;
    return __assign(__assign({ error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled !== null && disabled !== void 0 ? disabled : isSubmitting, onChange: onChange !== null && onChange !== void 0 ? onChange : function (date) {
            setFieldValue(field.name, date);
        }, onBlur: onBlur !== null && onBlur !== void 0 ? onBlur : function (e) {
            fieldOnBlur(e !== null && e !== void 0 ? e : field.name);
        }, onError: onError !== null && onError !== void 0 ? onError : createErrorHandler(fieldError, field.name, setFieldError) }, field), props);
}
function DateTimePicker(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(pickers.DateTimePicker, __assign({}, fieldToDateTimePicker(props)), children));
}
DateTimePicker.displayName = 'FormikMaterialUIDateTimePicker';

function fieldToKeyboardDateTimePicker(_a) {
    var disabled = _a.disabled, _b = _a.field, _onChange = _b.onChange, fieldOnBlur = _b.onBlur, field = __rest(_b, ["onChange", "onBlur"]), _c = _a.form, isSubmitting = _c.isSubmitting, touched = _c.touched, errors = _c.errors, setFieldValue = _c.setFieldValue, setFieldError = _c.setFieldError, onChange = _a.onChange, onBlur = _a.onBlur, onError = _a.onError, props = __rest(_a, ["disabled", "field", "form", "onChange", "onBlur", "onError"]);
    var fieldError = formik.getIn(errors, field.name);
    var showError = formik.getIn(touched, field.name) && !!fieldError;
    return __assign(__assign({ error: showError, helperText: showError ? fieldError : props.helperText, disabled: disabled !== null && disabled !== void 0 ? disabled : isSubmitting, onChange: onChange !== null && onChange !== void 0 ? onChange : function (date) {
            setFieldValue(field.name, date);
        }, onBlur: onBlur !== null && onBlur !== void 0 ? onBlur : function (e) {
            fieldOnBlur(e !== null && e !== void 0 ? e : field.name);
        }, onError: onError !== null && onError !== void 0 ? onError : createErrorHandler(fieldError, field.name, setFieldError) }, field), props);
}
function KeyboardDateTimePicker(_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(pickers.KeyboardDateTimePicker, __assign({}, fieldToKeyboardDateTimePicker(props)), children));
}
KeyboardDateTimePicker.displayName = 'FormikMaterialUIKeyboardDateTimePicker';

exports.fieldToDatePicker = fieldToDatePicker;
exports.DatePicker = DatePicker;
exports.fieldToKeyboardDatePicker = fieldToKeyboardDatePicker;
exports.KeyboardDatePicker = KeyboardDatePicker;
exports.fieldToTimePicker = fieldToTimePicker;
exports.TimePicker = TimePicker;
exports.fieldToKeyboardTimePicker = fieldToKeyboardTimePicker;
exports.KeyboardTimePicker = KeyboardTimePicker;
exports.fieldToDateTimePicker = fieldToDateTimePicker;
exports.DateTimePicker = DateTimePicker;
exports.fieldToKeyboardDateTimePicker = fieldToKeyboardDateTimePicker;
exports.KeyboardDateTimePicker = KeyboardDateTimePicker;
